src/platforms/wayland/displayclient.cpp

```diff
@@ -652,6 +652,12 @@ void mgw::DisplayClient::remove_global(
     // TODO: We should probably also delete any other globals we've bound to that disappear.
 }
 
+void mgw::DisplayClient::shell_ping(xdg_wm_base* shell, uint32_t serial)
+{
+    xdg_wm_base_pong(shell, serial);
+    wl_display_flush(display);
```

Comment:
Hm. Actually, this seems like the wrong place to flush? As per the [`wl_display_flush` documentation](https://wayland.freedesktop.org/docs/html/apb.html#Client-classwl__display_1a8463b6e5f4cf9a2a3ad2d543aedcf429), we should be calling it before waiting for events on the display - so, https://github.com/MirServer/mir/blob/7100296bea4de6f5c8e57733d7706e530abc59f2/src/platforms/wayland/display.cpp#L292 should be unconditional (and probably moved up to just before https://github.com/MirServer/mir/blob/7100296bea4de6f5c8e57733d7706e530abc59f2/src/platforms/wayland/display.cpp#L260 ?)

Reply:
Additionally, we should be handling failures from flush there.